R/conditions.R

```diff
@@ -0,0 +1,815 @@
+#' # condition.
+cnd_signal <- function(.cnd, ..., .msg = NULL, .call = FALSE,
+                       .mufflable = TRUE) {
+  call <- if (.call) sys.call(-1) else NULL
+  cnd <- make_cnd(.cnd, ..., .msg = .msg, .call = call)
+  cnd_signal_(cnd, base::signalCondition, .mufflable)
+}
+#' @rdname cnd_signal
+#' @export
+cnd_panic <- function(.cnd, ..., .msg = NULL, .call = FALSE,
+                      .mufflable = FALSE) {
+  call <- if (.call) sys.call(-1) else NULL
+  cnd <- make_cnd(.cnd, ..., .msg = .msg, .call = call)
+  cnd_signal_(cnd, base::stop, .mufflable)
+}
+
+make_cnd <- function(.cnd, ..., .msg, .call) {
+  if (is_scalar_character(.cnd)) {
+    .cnd <- cnd_new(.cnd, ...)
+  } else {
+    stopifnot(is_condition(.cnd))
+  }
+
+  # Override default fields if supplied
+  .cnd$message <- .msg %||% .cnd$message %||% ""
+  .cnd$call <- .call %||% .cnd$call
+
+  .cnd
+}
+cnd_signal_ <- function(cnd, signal, mufflable) {
+  if (mufflable) {
+    class(cnd) <- c("mufflable", class(cnd))
+    withRestarts(signal(cnd), muffle = function(...) NULL)
+  } else {
+    signal(cnd)
+  }
+}
+
+#' Signal an error, warning, or message.
+#'
+#' These functions are equivalent to base functions
+#' \code{\link[base]{stop}()}, \code{\link[base]{warning}()} and
+#' \code{\link[base]{message}()}, but the \code{type} argument makes
+#' it easy to create subclassed conditions. They also don't include
+#' call information by default. This saves you from typing
+#' \code{call. = FALSE} to make error messages cleaner within package
+#' functions.
+#'
+#' Like \code{stop()} and \code{\link{cnd_panic}()}, \code{panic()}
+#' signals a critical condition and interrupts execution by jumping to
+#' top level (see \code{\link{rst_abort}()}). Only a handler of the
+#' relevant type can prevent this jump by making another jump to a
+#' different target on the stack (see \code{\link{with_handlers}()}).
+#'
+#' \code{warn()} and \code{inform()} both have the side effect of
+#' displaying a message. These messages will not be displayed if a
+#' handler transfers control. Transfer can be achieved by establishing
+#' a thrown handler that transfers control to
+#' \code{\link{with_handlers}()}). In this case, the current function
+#' stops and execution resumes at the point where handlers were
+#' established.
+#'
+#' Since it is often desirable to continue normally after a message or
+#' warning, both \code{warn()} and \code{inform()} (and their base R
+#' equivalent) establish a muffle restart where handlers can jump to
+#' prevent the message from being displayed. Execution resumes
+#' normally after that. See \code{\link{rst_muffle}()} to jump to a
+#' muffling restart, and the \code{muffle} argument of
+#' \code{\link{inplace}()} for creating a muffling handler.
+#'
+#' @param msg A message to display.
+#' @param type Subclass of the condition to signal.
+#' @param call Whether to display the call.
+#'
+#' @export
+panic <- function(msg, type = NULL, call = FALSE) {
```

Comment:
That's how I called it at first until I understood that conditions are signalled in place rather than thrown in R. If the condition isn't handled (doesn't jump out of the signal function), then R jumps to the abort restart.

oh, how about `abort()`?